api/src/apps/routes.rs

```diff
@@ -513,4 +513,522 @@ mod tests {
             assert_eq!(run_options, Some(RunOptions::Sync));
         }
     }
+
+    mod url_rendering {
+        use crate::apps::AppsService;
+        use crate::infrastructure::Dummy;
+        use crate::models::{AppName, AppStatusChangeId};
+        use crate::sc;
+        use rocket::http::ContentType;
+        use rocket::http::Header;
+        use rocket::local::asynchronous::Client;
+        use std::convert::From;
+        use std::str::FromStr;
+        use std::sync::Arc;
+
+        use assert_json_diff::assert_json_include;
+        use serde_json::json;
+        use serde_json::Value;
+
+        #[tokio::test]
+        async fn host_header_response_with_xforwardedhost_xforwardedproto_and_xforwardedport(
+        ) -> Result<(), crate::apps::AppsServiceError> {
+            let (host_meta_cache, mut host_meta_crawler) = crate::host_meta_crawling();
+            let infrastructure = Box::new(Dummy::new());
+            let apps = Arc::new(AppsService::new(Default::default(), infrastructure).unwrap());
+            let _result = apps
+                .create_or_update(
+                    &AppName::from_str("master").unwrap(),
+                    &AppStatusChangeId::new(),
+                    None,
+                    &vec![sc!("service-a")],
+                )
+                .await?;
+            host_meta_crawler.fake_host_meta_info("master".to_string(), "service-a".to_string());
+            let rocket = rocket::build()
+                .manage(host_meta_cache)
+                .manage(apps)
+                .mount("/", routes![crate::apps::routes::apps])
+                .mount("/api/apps", crate::apps::apps_routes());
```

Comment:
This block appears to be the very same in a lot of tests. You could create a function for it.

```suggestion
            let rocket = set_up_rochet_with_dummy_infrastructure_and_a_running_app();
```